components/VM_Arm/src/main.c

```diff
@@ -856,7 +864,7 @@ static int load_vm(vm_t *vm, const char *kernel_name, const char *dtb_name, cons
 
     /* Attempt to load initrd if provided */
     guest_image_t initrd_image;
-    if (config_set(CONFIG_VM_INITRD_FILE)) {
+    if (vm_image_config.provide_initrd) {
```

Comment:
How about just `if (strlen(initrd_name))`? Then you could set `initrd_name` to empty string in CAmkES file to not load any.

Reply:
Yes, that would be sufficient to keep the config options lighter. Shoul we put this into https://github.com/seL4/camkes-vm/blob/296524002d908c475f364fcfbcbfa41496a5d2ee/components/VM_Arm/configurations/vm.h#L82 then:
```
string initrd_name = ""; /* no inird by default */ \
``` 
then as default in https://github.com/seL4/camkes-

Reply:
Sorry @chrisguikema for stealing the show, but let me show how I think this per-VM initrd is handled best: seL4/camkes-vm#59.

Reply:
Figured out we still have 3 special cases for guest's data blobs: kernel, DTB and initrd. I got an idea how to write that in a more generic way and remove lots of code from VM. Please give me a day or two so I can show my suggestion.

Reply:
I appreciate the review, Hannu, and I also appreciate that you're creating new MRs that address your comments